src/main/java/org/profit/telegrambot/controller/MainController.java

```diff
@@ -0,0 +1,122 @@
+            CustomerService.addCustomer(customer);
+            SendMessage sendMessage = new SendMessage();
+            sendMessage.setChatId(String.valueOf(message.getChatId()));
+            ComponentContainer.MY_TELEGRAM_BOT.sendMsg(sendMessage);
+            sendMessage.setReplyMarkup(new ReplyKeyboardRemove(true));
+        }
+
+        DeleteMessage deleteMessage = new DeleteMessage();
+        deleteMessage.setChatId(String.valueOf(message.getChatId()));
+        deleteMessage.setMessageId(message.getMessageId());
+        ComponentContainer.MY_TELEGRAM_BOT.sendMsg(deleteMessage);
+
+        SendMessage sendMessage = new SendMessage();
+        sendMessage.setText("Select an action:");
+        sendMessage.setChatId(String.valueOf(message.getChatId()));
+        sendMessage.setReplyMarkup(InlineKeyboardUtil.Menu());
+        ComponentContainer.MY_TELEGRAM_BOT.sendMsg(sendMessage);
+    }
+
+    private void handleText(Message message) {
+        String text = message.getText();
+
+        SendMessage sendMessage = new SendMessage();
+        sendMessage.setChatId(String.valueOf(message.getChatId()));
+        Customer customer = CustomerService.getCustomerById(String.valueOf(message.getChatId()));
+
+        if (text.equals("/start")) {
+
+            if (customer == null) {
+                sendMessage.setText("<b>Hey!</b>\n"
+                        + "Send me your number.");
+                sendMessage.setParseMode(ParseMode.HTML);
+                sendMessage.setReplyMarkup(KeyboardUtil.contactMarkup());
+                sendMessage.setChatId(String.valueOf(message.getChatId()));
+                ComponentContainer.MY_TELEGRAM_BOT.sendMsg(sendMessage);
+
+            } else {
+                sendMessage.setText("<b>💻AHello and welcome to JabkoBot ! \n\nSelect One from the menu</b>\n");
+                sendMessage.setParseMode(ParseMode.HTML);
+
+                sendMessage.setReplyMarkup(InlineKeyboardUtil.Menu());
+                ComponentContainer.MY_TELEGRAM_BOT.sendMsg(sendMessage);
+            }
+        }
+    }
+
+
+    public void handleCallBack(Message message, String data) {
+
+        if (data.equals("web_network")) {
+
+            DeleteMessage deleteMessage = new DeleteMessage();
+            deleteMessage.setChatId(String.valueOf(message.getChatId()));
+            deleteMessage.setMessageId(message.getMessageId());
+            ComponentContainer.MY_TELEGRAM_BOT.sendMsg(deleteMessage);
+
+            SendMessage sendMessage = new SendMessage();
+            sendMessage.setText("Our networks");
+            sendMessage.setChatId(String.valueOf(message.getChatId()));
+            sendMessage.setReplyMarkup(InlineKeyboardUtil.NetAddressMenu());
+            ComponentContainer.MY_TELEGRAM_BOT.sendMsg(sendMessage);
+
+        } else if (data.equals("back")) {
+            DeleteMessage deleteMessage = new DeleteMessage();
```

Comment:
Here is just for your information:

Seems there are a certain number of the same code over there, right?
Actually, that is a bad practice. There is even a programming principle DRY that stands for **Do not Repeat Yourself**.

Meanwhile there are a few ways to refactor that (extract to some method), I would reccomend to go through https://refactoring.guru/uk, namely - https://refactoring.guru/uk/design-patterns/builder. 

That guy could harmonize here, but who am I to state 😁?

PS: other patterns are also worth to be familiar with